src/migrations/user-store/5.0.1.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { app } from "electron";
+import { existsSync, readJsonSync } from "fs-extra";
+import path from "path";
+import os from "os";
+import { ClusterStore, ClusterStoreModel } from "../../common/cluster-store";
+import type { KubeconfigSyncEntry, UserPreferencesModel } from "../../common/user-store";
+import { MigrationDeclaration, migrationLog } from "../helpers";
+
+export default {
+  version: "5.0.1",
+  run(store) {
+    const { syncKubeconfigEntries = [], ...preferences }: UserPreferencesModel = store.get("preferences") ?? {};
+    const { clusters = [] }: ClusterStoreModel = readJsonSync(path.resolve(app.getPath("userData"), "lens-cluster-store.json"));
+    const syncPaths = new Set(syncKubeconfigEntries.map(s => s.filePath));
+
+    syncPaths.add(path.join(os.homedir(), ".kube"));
```

Comment:
Does this get executed even for a new install of Lens?

Reply:
Yes because the file doesn't exist yet

Reply:
So migrations get run even for a new install? Adding ".kube" to the list of sync'ed paths is "setup" and it seems odd to have it done in a migration (and not easy to find)

Reply:
In light of #3254 should we just add ".kube/config"? User can manually add ".kube" in that case. But as it is the user cannot remove ".kube" (because it will keep getting added back, right?) Using ".kube" folder to save backups is probably not a typical use case but certainly possible. Could be done in another PR.

Reply:
That was the intentional design. I would actually choose to close that issue as "working as intended"

Reply:
FYI this migration will only happen once. So once this migration happens a user will be able to remove `~/.kube` to their hearts desire.